### PR TITLE
Add statistics caching

### PR DESCRIPTION
**Pull request by WebDevelopUa**
Adds caching mechanism for statistics for local development
without internet connection or for the cases when API is down.

### CODE REVIEW

**Comment by viktor-yakubiv** [comment]
@samuelpearce this is something you may be interested in.

For this moment on each build we query our API to retrieve statistics and then blend it into HTML. This PR changes such behaviour to the following:
- use cached result from file if the file exists and was not modified since last 6 hours
- use API result if network responds (and save it to cache)
- use default file with some numbers otherwise

This kind of behaviour is enabled for development only, so builds for both testing and production will fails. I assume that we use clean workspace in both environments before build so it would be tricky to use some kind of cache for API is down case.

I am double checking with you @samuelpearce is behaviour after this PR correct?

File: webpack/data-loader.js [hunk 1]
```diff
@@ -20,10 +30,42 @@ const retrieveStats = url => {
...
+  if (!ignoreModified) {
+    const cacheStats = await fsPromises.stat(cacheFilePath)
+    if (cacheStats.mtimeMs < Date.now() - 6 * 60 * 60 * 1000)
+      throw new Error('Cache outdated')
+  }
+
+  return fsPromises
+    .readFile(cacheFilePath)
+    .then(contents => JSON.parse(contents))
+}
```
Comment (Joozty) [inline webpack/data-loader.js]: Maybe this one would look better, wouldn't it?

```
  const contents = await fsPromises.readFile(cacheFilePath)
  return JSON.parse(contents)
```

File: webpack/data-loader.js [hunk 2]
```diff
@@ -20,10 +30,42 @@ const retrieveStats = url => {
...
+
+const retrieveStats = async (url, catchFilePath) => {
+  try {
+    return await loadCachedStats(catchFilePath)
+  } catch (cacheError) {
+    try {
+      const stats = await fetchStats(url)
+      saveCachedStats(catchFilePath, JSON.stringify(stats))
```
Comment (Joozty) [inline webpack/data-loader.js]: It should be saved only when  process.env.NODE_ENV !== 'production'.

File: webpack/data-loader.js (hunk 1 above)
Comment (viktor-yakubiv) [reply webpack/data-loader.js]: Maybe even `fsPromises.readFile(cacheFilePath).then(JSON.parse)` would look simpler

**Pull request merged**